### DISCUSSION TITLE
Would it make sens to move DomainObjectInterface to symfony/contracts ?

### DISCUSSION THREAD

**Discussion by VincentLanglet** [symfony/symfony#48257] [Q&A]
symfony/security-acl is not recommended for most situation anymore so even if some libraries can provide acl supports, it would always be better to have an optional support instead of having this as a required dependency.
(https://github.com/sonata-project/SonataAdminBundle/issues/7303 for context).

When working with "domain objects which you want to store ACLs" there is a `DomainObjectInterface` to "return a unique identifier for this domain object".

Let's say I have
- A domain object related to some security stuff
- A security service implementing some SecurityInterface without using ACL (1)
- A security service implementing some SecurityInterface using ACL (2)

In this example the domain object is shared between both SecurityInterface implementation. Even if I can load dynamically the service config whether or not the `symfony/security-acl` is installed, the domain object will still need to implement the `DomainObjectInterface` so it cannot be a fully optional dependency...

Then would it make sens to move DomainObjectInterface to symfony/contracts ? This way
- symfony/acl-contracts would be required
- The domain object implements `DomainObjectInterface` from `symfony/acl-contracts`
- If symfony/security-acl is not installed, the services (1) are loaded.
- If symfony/security-acl is installed, both the services (1) and (2) are loaded.

Or am I missing something in the way to move a dependency from require to optional ?

**Comment by stof** [comment]
The recommended solution for authorization is to use custom voters rather than ACLs. So I doubt we will create a new contracts package for it.

**Reply by VincentLanglet** [reply]
Does it mean ACL will be deprecated one day ?

I personally use custom voters, but when open-sourcing a library it might be useful to support both.
And it would help supporting ACL without requiring the dependency.

If ACL won't be deprecated, new contract might make sens.
If ACL will be deprecated any time soon, that's different.

**Reply by stof** [reply]
They're not officially deprecated for now (and we haven't done an explicit plan for them) but they are officially discouraged and they are considered community-maintained (they don't receive the same attention from the core team than other components).
So to me, when open-sourcing a library, I would not care about ACLs.
